pkg/sysregistriesv2/alias.go

```diff
@@ -0,0 +1,112 @@
+package sysregistriesv2
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/BurntSushi/toml"
+	"github.com/containers/image/v5/types"
+	"github.com/containers/storage/pkg/homedir"
+	"github.com/containers/storage/pkg/lockfile"
+)
+
+// userAliasesFile is the user-specific registries.conf file to store aliases.
+var userAliasesFile = filepath.FromSlash(".config/containers/aliases.conf")
+
+// userAliasesLockFile is the lockfile for the user-specific registries.conf
+// file to store aliases.
+var userAliasesLockFile = filepath.FromSlash(".local/share/containers/aliases.conf.lock")
+
+// ResolveAlias performs an alias resolution of the specified name.  The
+// user-specific aliases.conf has precedence over aliases in `conf`.
+func (conf *V2RegistriesConf) ResolveAlias(ctx *types.SystemContext, name string) (string, bool, error) {
+	confPath, lock, err := aliasesConfPathAndLock(ctx)
+	if err != nil {
+		return "", false, err
+	}
+
+	// Acquire the lock as a reader to allow for multiple routines in the
+	// same process space to read in simultaneously.
+	lock.RLock()
+	defer lock.Unlock()
+
+	aliasConf, err := loadAliasesConf(confPath)
+	if err != nil {
+		return "", false, err
+	}
+
+	// Note: aliases.conf has precedence.
+	value, resolved := aliasConf.Aliases[name]
+	if !resolved {
+		value, resolved = conf.Aliases[name]
+	}
+	return value, resolved, nil
+}
+
+// AddAliases adds the specified name-value pair as a new alias to the
+// user-specific aliases.conf.  It may override an existing alias for `name`.
+func (conf *V2RegistriesConf) AddAlias(ctx *types.SystemContext, name string, value string) error {
```

Comment:
Same here:
- Make this a top-level function
- Consider making the `value` a `reference.Named`.

… and eventually probably validate the syntax of both the alias and the resolved value before writing it to the file.

Reply:
Good catch. I also just felt into the trap when working on `pkg/shortnames`.